src/components/Navigator.js

```diff
@@ -212,13 +263,31 @@ class Navigator extends React.Component {
         ? storedInitialFileId.sourceId : NO_OPEN_FILE_ID.sourceId,
       viewId: storedInitialFileId.hasOwnProperty('viewId') ? storedInitialFileId.viewId : NO_OPEN_FILE_ID.viewId,
     }
-    if (
-      initialFileId.sourceId !== this.props.currentOpenFileId.sourceId ||
-      initialFileId.viewId !== this.props.currentOpenFileId.viewId
-    ) { handleSetCurrentOpenFileId(initialFileId); }
+    doGetFilesList().then(filesList => {
+      if (!filesList) { alert('failed to retrieve files list'); }
+      else { this.setState({ filesList }); }
+      if (this.props.backendModeSignedInStatus === BACKEND_MODE_SIGNED_IN_STATUS.LOCAL_STORAGE) {
+        this.setState({ nextNewFileIds: calculateLocalStorageNextNewFileIds(this.state.filesList) });
+      }
+      if (
+        initialFileId.sourceId !== this.props.currentOpenFileId.sourceId ||
+        initialFileId.viewId !== this.props.currentOpenFileId.viewId
+      ) { handleSetCurrentOpenFileId(initialFileId); }
+    });
```

Comment:
load this in componentDidMount instead of the constructor since the async nature of doGetFilesList meant that setting the state in the constructor in the then block happens after setting the initial file id, which results in some bad behavior when opening that file. Settin the initial file id has to be done in the mount since the handleSetCurrentOpenFileId method causes issues in the constructor.